README.md

```diff
@@ -0,0 +1,90 @@
+# EVT-core
+
+## Introduction
+
+### What is EVT?
+EVT stands for the Electric Vehicle Team at RIT. We build and race electric
+vehicles. Currently we have two track motorcycles we develop for and are
+desiging an electric dirt bike.
+
+This repository is managed by the Firmware team which handles the firmware
+used by the team on the various vehicles.
+
+### What is EVT-Core?
+EVT-Core is the central library used by EVT for interacting with the MCUs on
+the different components on the electric vehicles. The library is designed to
+be an alternative to Mbed which was the previous tool used for programming the
+MCUs. EVT-Core is not a RTOS, it instead contains abstractions for bare metal
+programming. Contained is abstractions for working with I2C, SPI, GPIO, PWM,
+RTC, etc.
+
+### How is EVT-Core Used?
+EVT-Core is a CMake library and as such can be built as a library and
+manually linked to, or can be added as a sub-directory to a larger project.
+Further information is to come.
+
+### How do I Build the Code?
+The compilation process is handled via CMake and in general follows standard
+practices. The noticable addition is the need to setup an environment variable
+which stores the location to the gcc arm toolchain. Below is an example of
+how to set the environment variable on Linux/Unix.
+
+```bash
+export GCC_ARM_TOOLS_PATH=/path/to/gcc_arm_tools_path/bin/
+```
+
+The compilation logic will make use of that enivronment variable to locate
+the needed tools to generate the code.
+
+After that, the compilation process follows standard practice. Below is
+an example compilation process.
+
+```bash
+mkdir build/
+cd build/
+cmake ../
+make
```

Comment:
For a parallel build:
```suggestion
make -j $(nproc)
```

Reply:
For a build machine it's probably fine to just use -j to allow make to decide.